### ISSUE TITLE
Allow fitting models on raster stacks of differing extents

### ISSUE THREAD

**Issue by ailich** [bleutner/RStoolbox#82]
Sometimes I'd like to fit a single model using `superClass` or `unsuperClass` on several disjointed areas where I have raster data of the same variables. The way I've done this is the past for supervised classification is to sample each raster at the training points to create a single dataframe, then fit the model in `caret`, and the use `raster::predict` on each raster individually. It would be nice however if this could be done directly in `RSToolbox` and perhaps with the upgrade to `terra` you could include [`SpatRasterDatasets`](https://rdrr.io/github/rspatial/terra/man/sds.html) as a supported input which is essentially like having a `list` of rasters/raster stacks that can be of differing extents.

**Comment by bleutner** [comment]
makes sense, yes. I've been reworking large parts of superClass and will consider this suggestion. Also SpatRasterDatasets seems like a good input structure I wasn't aware of.

**Comment by ailich** [comment]
Thanks!

**Comment by ailich** [comment]
I was actually mistaken about the differing extents. I tried using a `SpatRasterDataset` instead of a `list` of rasters and got "Error: [sds] extents do not match ()".
